### ISSUE TITLE
No stack trace when compiling with option `-g'

### ISSUE THREAD

**Issue by dhil** [ocamllabs/ocaml-multicore#59]
The following program `false.ml`:

```
let _  = assert false
```

does not admit a stack trace when compiled using the debug option `-g`, e.g.

```
$ ocamlopt -v
The OCaml native-code compiler, version 4.02.2+multicore-dev0
Standard library directory: /home/dhil/.opam/4.02.2+local-git-fix-sys-threads/lib/ocaml
$ ocamlopt -g false.ml
$ ./a.out
Fatal error: exception Assert_failure("false.ml", 1, 8)
```

Although, a stack trace is produced when compiled with the 4.02.3 OCaml compiler:

```
$ ocamlopt -v 
The OCaml native-code compiler, version 4.02.3
Standard library directory: /home/dhil/.opam/4.02.3/lib/ocaml
$ ocamlopt -g false.ml
$ ./a.out
Fatal error: exception Assert_failure("false.ml", 1, 8)
Raised at file "false.ml", line 1, characters 8-20
```

**Comment by kayceesrk** [comment]
Related to #93.

**Comment by kayceesrk** [comment]
Fixed by #106.
